tests/fixtures/scenarios/exitCode/components/terraform/hook-and-store/main.tf

```diff
@@ -0,0 +1,31 @@
+terraform {
+  required_providers {
+    null = {
+      source  = "hashicorp/null"
+      version = "~> 3.0"
+    }
+  }
+}
+
+variable "stage" {
+  description = "Stage. Used to define an Atmos stack."
+  type        = string
+  default     = "test"
+}
+
+variable "exit_code" {
+  description = "Exit code used to simulate the command's exit status."
+  type        =  number
+  default     =  0
+}
+
+
+resource "null_resource" "fail_on_second_apply" {
+  triggers = {
+    always_run = timestamp()
+  }
+
+  provisioner "local-exec" {
+    command = "exit ${var.exit_code}"
```

Comment:
Ok, this works for me, but we're calling this component `hook-and-store`. 

Instead, let's call the component just `exit-code` and make it a shared component.

Reply:
Done